sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/ExpressionEncoder.scala

```diff
@@ -191,7 +191,7 @@ object ExpressionEncoder {
    * thread-safe. Note that multiple calls to `apply(..)` return the same actual [[InternalRow]]
    * object.  Thus, the caller should copy the result before making another call if required.
    */
-  class Serializer[T](private val expressions: Seq[Expression])
```

Comment:
Please avoid this change.
You can use `ExpressionEncoder.createDeserializer`

Reply:
Thank you for reminding this @beliefer , it was modified by accident and I have reverted it, PTAL again.